cdap-ui/app/features/hydrator/services/pipeline-error-factory.js

```diff
@@ -166,31 +165,19 @@ angular.module(PKG.name + '.feature.hydrator')
       return plugin.valid;
     }
 
-    function countRequiredFields(plugin) {
+    function countRequiredFields(node) {
       var requiredFieldCount = 0;
-      var i;
-      var keys;
-      if (angular.isObject(plugin._backendProperties) && Object.keys(plugin._backendProperties).length) {
-        keys = Object.keys(plugin._backendProperties);
-        if (angular.isObject(plugin.properties)) {
-          if (!Object.keys(plugin.properties).length) {
-            for (i =0; i<keys.length; i++) {
-              if (plugin._backendProperties[keys[i]] && plugin._backendProperties[keys[i]].required) {
-                requiredFieldCount += 1;
-              }
-            }
-            return requiredFieldCount;
-          } else {
-            for (i=0; i< keys.length; i++) {
-              var property = plugin.properties[keys[i]];
-              if (plugin._backendProperties[keys[i]] && plugin._backendProperties[keys[i]].required && (!property || property === '')) {
-                requiredFieldCount += 1;
-              }
+      if (angular.isObject(node._backendProperties) && Object.keys(node._backendProperties).length) {
+
+        angular.forEach(node._backendProperties, function (value, key) {
+          if (value.required) {
+            if (!node.plugin.properties || !node.plugin.properties[key] || node.plugin.properties[key] === '') {
```

Comment:
the third condition of this isn't required, right? (its implicit in the second?)